src/main/kotlin/de/code_freak/codefreak/service/ContainerService.kt

```diff
@@ -144,4 +150,39 @@ class ContainerService(
 
   protected fun isContainerRunning(containerId: String): Boolean =
       docker.inspectContainer(containerId).state().running()
+
+  @Transactional
+  @Scheduled(fixedRate = SHUTDOWN_TASK_RATE, initialDelay = SHUTDOWN_TASK_RATE)
+  protected fun shutdownIdleIdeContainers() {
+    log.debug("Checking for idle containers")
+    // create a new map to not leak memory if containers disappear in another way
+    val newIdleContainers: MutableMap<String, Long> = mutableMapOf()
+    docker.listContainers(
+        DockerClient.ListContainersParam.withLabel(LABEL_ANSWER_ID),
+        DockerClient.ListContainersParam.withStatusRunning())
+        .forEach {
+          val containerId = it.id()
+          // TODO: Use `cat /proc/net/tcp` instead of lsof (requires no privileges)
+          val connections = exec(containerId, arrayOf("/opt/code-freak/num-active-connections.sh")).trim()
+          if (connections == "0") {
+            val idleTime = idleContainers[containerId] ?: 0
+            log.debug("Container $containerId has been idle for more than $idleTime ms")
+            if (idleTime >= SHUTDOWN_IDLE_THRESHOLD) {
+              val answerId = it.labels()!![LABEL_ANSWER_ID]
+              log.info("Shutting down container $containerId of answer $answerId")
+              val answer = answerRepository.findById(UUID.fromString(answerId))
+              if (answer.isPresent) {
+                saveAnswerFiles(answer.get())
```

Comment:
This fails for me with the following exception: https://gist.github.com/HenningCash/df9ae3fd1029d43f4ecd7e4020e6c780

Reply:
Should be fixed by a324909